### ISSUE TITLE
key/referrer mismatch error thrown

### ISSUE THREAD

**Issue by chris-aeviator** [matterport/sdk-starter-kit#5]
When setting up the sdk-starter-kit as described in your readme, I get a `key/referrer mismatch` thrown in my browser's console.

Steps to reproduce:

- clone the sdk
- run `yarn install`
- input Sid (JHK1JdUZpg2&qs) & API key (starts with eb86f8) into `config.json`
- start dev server by running `yarn run debug`
- visit the site by browsing to localhost:80 (changed the port to 8080 in `webpack.config.js` for obvious reasons)
- open Browser console

full browser console output:
![image](https://user-images.githubusercontent.com/11522213/44395697-f7dde200-a53a-11e8-8082-4b53c2df4c00.png)

Analytics call is blocked through network wide adblock...

**Comment by npeditto** [comment]
I noticed the same...

**Comment by mmckeaveney** [comment]
Seeing the same issue. Was anyone able to get the SDK working at all?

**Comment by chris-aeviator** [comment]
There's also no reaction from matterport support (email) for this since about 3 weeks now :-1:

Seem like this error means that your key is invalid (as is ours, without ever beeing officially rejected/notified), since I can trigger the same error with a random API key.

**Comment by mmckeaveney** [comment]
@chris-aeviator The problem for me was that the URL my application was served on did not match the matterport API keys matching referrer URL. If you look at the response in the call to `/v1/user`, you should be able to see what isn't matching. Even if the port is different, this will not work. For example - if you are serving over `localhost:4000` and your dev key is set up to use `localhost`, you will need to run your server on port 80 for it to work.

**Comment by chris-aeviator** [comment]
if you experience port 80 not serving correct make sure to run with `sudo` as in `sudo yarn debug`

**Comment by Alexuisis** [comment]
same here.. and the funny this is that I was following the official documentation with the account key.

**Comment by cheahph** [comment]
is the issue being resolved? I have the same problem also..

**Comment by chris-aeviator** [comment]
@cheahph make sure you run it on port 80, that solved the issue for me

**Comment by Alexuisis** [comment]
> is the issue being resolved? I have the same problem also..

you need to have paid subscription for this.

**Comment by cheahph** [comment]
I have the paid matterport subscription but it is still not working..it gives me the " Failed to get User data" and ""This referrer is not allowed to use this X-Matterport-Application-Key"..probably you need to pay to use the sdk also?

**Comment by chris-aeviator** [comment]
This is not a subscription error. Referer refers to the URL. 

According to the docs the sdk only works on

- localhost (NOT 127.0.0.1, NOT 0.0.0.0, NOT a domain example.com)
- port 80

So biggest question - what's in your URL bar?

----------

**Comment by gusi1994** [comment]
@chris-aeviator  I have the save issue as @cheahph. I tried running this starter-kit on localhost port 80 but no success. The save issue arrises even if I use code from from the documentation. 

Is a production key for the bundle sdk needed to get this working? Even on Localhost?

Because another issue that happens a lot is that XMLHttpRequests get blocked by CORS policy from https://my.matterport.com/api/v1/user even the fields x-matterport-application-key and x-matterport-referrer have the right values inside the request header.

Has anyone a solution?

**Comment by chris-aeviator** [comment]
> @chris-aeviator I have the save issue as @cheahph. I tried running this starter-kit on localhost port 80 but no success. The save issue arrises even if I use code from from the documentation.
> 
> Is a production key for the bundle sdk needed to get this working? Even on Localhost?
> 
> Because another issue that happens a lot is that XMLHttpRequests get blocked by CORS policy from https://my.matterport.com/api/v1/user even the fields x-matterport-application-key and x-matterport-referrer have the right values inside the request header.
> 
> Has anyone a solution?

Yes you need the SDK key from matterport

**Comment by gusi1994** [comment]
Thanks @chris-aeviator for the info. I almost thought so. The documentation says that demo models can be used with a sandbox key on localhost. But the model ID from the SDK samples doesn't work either because CORS errors occur.

I think I have to wait until they anwser my inquiry.
